cpp/src/arrow/compute/kernels/vector_run_end_encode.cc

```diff
@@ -398,17 +409,19 @@ class RunEndDecodeImpl {
       }
     }
 
-    ARROW_ASSIGN_OR_RAISE(auto output_array_data,
-                          ree_util::PreallocateValuesArray(
-                              ree_type->value_type(), has_validity_buffer, length,
-                              kUnknownNullCount, ctx_->memory_pool(), data_buffer_size));
+    ARROW_ASSIGN_OR_RAISE(
+        auto output_array_data,
+        ree_util::PreallocateValuesArray(ree_type->value_type(), has_validity_buffer,
+                                         length, ctx_->memory_pool(), data_buffer_size));
 
     int64_t output_null_count = 0;
     if (length > 0) {
       RunEndDecodingLoop<RunEndType, ValueType, has_validity_buffer> loop(
           input_array_, output_array_data.get());
       output_null_count = length - loop.ExpandAllRuns();
     }
+    DCHECK(output_array_data->null_count == kUnknownNullCount ||
+           output_array_data->null_count == 0);
```

Comment:
Is this useful?
It seems that this just checks a return value of `PreallocateValuesArray()`. If this is a post-condition of `PreallocateValuesArray()`, how about doing this in `PreallocateValuesArray()` not in callers?

Reply:
Same as above. I'm removing the DCHECK to reduce the noise here.